python/lsst/ts/mthexapod/structs.py

```diff
@@ -66,26 +85,53 @@ class Telemetry(ctypes.Structure):
 
     _pack_ = 1
     _fields_ = [
+        # extBissEncoderRaw in telemetryStreamStructure_t
+        ("strut_measured_pos_raw", ctypes.c_int32 * 6),
         ("status_word", ctypes.c_uint16 * 6),
         ("latching_fault_status_register", ctypes.c_uint16 * 6),
         ("copley_fault_status_register", ctypes.c_uint32 * 6),
         ("application_status", ctypes.c_uint32),
         ("input_pin_states", ctypes.c_uint32 * 3),
-        ("motor_current", ctypes.c_double * 6),
-        ("motor_voltage", ctypes.c_double * 6),
+        # TODO DM-31290: uncomment these lines and move them
+        # to the correct location when the data is available
+        # ("motor_current", ctypes.c_double * 6),
+        # ("motor_voltage", ctypes.c_double * 6),
+        ("mjd", ctypes.c_double),
+        ("mjd_frac", ctypes.c_double),
+        ("dt", ctypes.c_double),
+        # StrutPosition_1 in telemetryStreamStructure_t;
+        # documented as: simulink "calibrated" position (in microns?)
+        ("strut_measured_pos_um", ctypes.c_double * 6),
+        # CommandPos_n, CommandVel_n in telemetryStreamStructure_t
+        ("strut_commanded_posvel", PosVel * 6),
+        # CommandAccel_n in telemetryStreamStructure_t
+        ("strut_commanded_accel", ctypes.c_double * 6),
+        # CommandFinalPos_n in telemetryStreamStructure_t
+        ("strut_commanded_final_pos", ctypes.c_double * 6),
+        ("command_valid", ctypes.c_double),
+        ("track_mode", ctypes.c_double),
         # simulink telemetry
         ("state", ctypes.c_double),
         ("enabled_substate", ctypes.c_double),
         ("offline_substate", ctypes.c_double),
         ("test_state", ctypes.c_double),
-        ("strut_encoder_raw", ctypes.c_double * 6),
-        ("strut_encoder_microns", ctypes.c_double * 6),
-        # commanded displacement (microns) and rotation (deg)
+        # PosErrorn in telemetryStreamStructure_t with this note:
+        # Get the PosError from the drive directly instead of Simulink Model
+        ("strut_pos_error", ctypes.c_double * 6),
+        # Rn_Est in telemetryStreamStructure_t
+        ("measured_uvw", ctypes.c_double * 3),
+        # Posn_Est in telemetryStreamStructure_t
+        ("measured_xyz", ctypes.c_double * 3),
+        ("num_iterations", ctypes.c_double),
+        # EstPos_M_n, EstPos_M_Filt_n, EstVel_n in telemetryStreamStructure_t
+        ("estimated_posfiltvel", PosFiltVel * 6),
+        # LinearEncodern in telemetryStreamStructure_t;
+        # I have no idea how this differs from strut_measured_pos_raw
+        ("strut_linear_encoder", ctypes.c_double),
+        # LinearEncoderVelocityn in telemetryStreamStructure_t
+        ("strut_linear_encoder_velocity", ctypes.c_double),
```

Comment:
There should be 6 fields:
https://github.com/lsst-ts/ts_hexapod_controller/blob/develop/include/interface/telemetryStructure.h#L159-L164

Reply:
Thank you for catching that. After fixing it I find that the struct has 552 bytes (excluding the header). Does that sound right?

Also is input_pin_states three uint32 values (one per pair of struts) or 6 uint16 values (one per strut)? It is listed as 3 x uint32 but there is a comment for that field which suggests it should be 6 x uint16.

Reply:
I think you should have `88 + 656 + 8*6 = 792` bytes if there is no header.

For the `input_pin_states`, it should be: `uint32_t input_pin_states[3]`. One drive (contains two actuators) gives one input pin state. There are 3 drives.

Reply:
I found the problem. I had `_fields` instead of `_fields_` (missing trailing underscore) in  my definitions of the PosVel and PosFiltVel structs. As a result they had no fields and size = 0. With that fixed I get a size of 792 bytes (including the commanded position that you will be adding), as expected.